### ISSUE TITLE
Param store loading / saving does not load/save registered buffers

### ISSUE THREAD

**Issue by gizatt** [pyro-ppl/pyro#1839]
I ran into some relatively confusing behavior when trying to save / load a model containing an AutoregressiveNN -- I'd train up my model, evaluate it, and save it to disk, but once I loaded it back, its behavior would change. The reason is that it includes a MaskedLinear layer with a randomly-generated (by default) mask, which is registered to PyTorch with `register_buffer`, but not saved out with the Pyro param-store-saving machinery.

I'm not sure what the path forward for (or necessity of) resolution is -- it's easy for me to patch what I'm doing now that I understand it by being careful with random seeds or otherwise supplying my own deterministic permutation. (And maybe refactoring my code so I can use pytorch's state-dict save/load on its own.)  

But perhaps Pyro ought to keep track of "fixed parameters" as well during during its own parameter saving (differentiating between optimized and fixed parameters) -- unless there's already a pleasant way of saving and loading both Pyro params + the remaining underlying Pytorch state dict?

**Comment by martinjankowiak** [comment]
there's an example usage that mixes pyro's and pytorch save/load functionality in the dmm tutorial:

http://pyro.ai/examples/dmm.html#Checkpointing

i'm not really sure what the best resolution here is. it's probably best if the user is responsible for saving/loading torch modules. we probably don't want things to be too automagical here.
